Core/GDCore/Project/Variable.cpp

```diff
@@ -66,100 +152,111 @@ const Variable& Variable::GetChild(const gd::String& name) const {
   auto it = children.find(name);
   if (it != children.end()) return *it->second;
 
-  isStructure = true;
+  type = Type::Structure;
   children[name] = std::make_shared<gd::Variable>();
   return *children[name];
 }
 
 void Variable::RemoveChild(const gd::String& name) {
-  if (!isStructure) return;
+  if (type != Type::Structure) return;
   children.erase(name);
-  isStructure = !children.empty();
 }
 
 bool Variable::RenameChild(const gd::String& oldName,
                            const gd::String& newName) {
-  if (!isStructure || !HasChild(oldName) || HasChild(newName)) return false;
+  if (type != Type::Structure || !HasChild(oldName) || HasChild(newName))
+    return false;
 
   children[newName] = children[oldName];
   children.erase(oldName);
 
   return true;
 }
 
-void Variable::ClearChildren() {
-  if (!isStructure) return;
-  children.clear();
-}
+Variable& Variable::GetAtIndex(const size_t index) {
+  type = Type::Array;
+  while (childrenList.size() <= index)
+    childrenList.push_back(std::make_shared<gd::Variable>());
+  return *childrenList[index];
+};
+
+const Variable& Variable::GetAtIndex(const size_t index) const {
+  if (childrenList.size() <= index) return *std::make_shared<gd::Variable>();

```

Comment:
I still have a lot to learn about c++, what a complex language. I thought that as I was initialising on the heap and using a smart pointer, it would be working fine :/
Anyways, a static badVariable is a cleaner and more memory friendly solution either way.